UI/src/javascript/tuition-partner.js

```diff
@@ -0,0 +1,109 @@
+"use strict";
+
+import axios from "axios";
+
+const TuitionPartner = (() => {
+  const getSeoUrlFromId = (id) => id && id.replace("shortlist-tpInfo-cb-", "");
+  const getAddToShortlistFetchUrl = (id) =>
+    `/tuition-partner/${id}?handler=AddToShortlist`;
+  const getRemoveFromShortlistFetchUrl = (id) =>
+    `/tuition-partner/${id}?handler=RemoveFromShortlist`;
+
+  const shortlistCheckboxes = document.querySelector(
+    '[data-module="tuition-partner-shortlist"]'
+  );
+
+  const isResultValid = (result) =>
+    result && result.updated !== "undefined" && result.updated;
+  const throwError = (message = "Invalid result") => {
+    throw new Error(message);
+  };
+
+  const init = () => {
+    if (shortlistCheckboxes) {
+      shortlistCheckboxes
+        .querySelector("input[type='checkbox']")
+        .addEventListener("click", onCheckboxClick);
+    }
+  };
+  const onCheckboxClick = async (event) => {
```

Comment:
Most of this logic is basically the same as `onCheckboxClick` for the search results. Could we combine these so we have a single place where the API is called?

Reply:
Thanks for the suggestion. Sure, I will look into that.

Reply:
done